app/controllers/api/v1/accounts/lookup_controller.rb

```diff
@@ -3,6 +3,8 @@
 class Api::V1::Accounts::LookupController < Api::BaseController
   before_action -> { authorize_if_got_token! :read, :'read:accounts' }
   before_action :set_account
+  before_action :check_account_approval
```

Comment:
These changes would mean this API is no longer a reliable way to validate username availability on sign-up

Reply:
Oh, I overlooked that use case for that API. I'm going to change that.